src/bin/pg_autoctl/cli_enable_disable.c

```diff
@@ -1200,3 +1228,540 @@ cli_disable_ssl(int argc, char **argv)
 
 	(void) cli_enable_ssl(argc, argv);
 }
+
+
+/*
+ * cli_enable_monitor_getopts parses the command line options for the
+ * command `pg_autoctl enable monitor`.
+ */
+static int
+cli_enable_monitor_getopts(int argc, char **argv)
+{
+	KeeperConfig options = { 0 };
+	int c, option_index = 0, errors = 0;
+	int verboseCount = 0;
+
+	static struct option long_options[] = {
+		{ "pgdata", required_argument, NULL, 'D' },
+		{ "monitor", required_argument, NULL, 'm' },
+		{ "allow-failover", no_argument, NULL, 'A' },
+		{ "version", no_argument, NULL, 'V' },
+		{ "verbose", no_argument, NULL, 'v' },
+		{ "quiet", no_argument, NULL, 'q' },
+		{ "help", no_argument, NULL, 'h' },
+		{ NULL, 0, NULL, 0 }
+	};
+
+	/* set default values for our options, when we have some */
+	options.groupId = -1;
+	options.network_partition_timeout = -1;
+	options.prepare_promotion_catchup = -1;
+	options.prepare_promotion_walreceiver = -1;
+	options.postgresql_restart_failure_timeout = -1;
+	options.postgresql_restart_failure_max_retries = -1;
+
+	options.pgSetup.settings.candidatePriority =
+		FAILOVER_NODE_CANDIDATE_PRIORITY;
+	options.pgSetup.settings.replicationQuorum =
+		FAILOVER_NODE_REPLICATION_QUORUM;
```

Comment:
It turns out that we can't remove them yet, because otherwise we have candidate priority 0 and replication quorum false. Those values are not kept in the local state nor local configuration of the node, only managed on the monitor, to avoid “cache invalidation” problems. It means we can't recover the previous settings after having lost the monitor.

We can either leave things as they are now: reuse default values when re-joining nodes, but then edit the documentation about that, or make it so that we register the replication settings in the local state/config and recover them when re-joining a new monitor. We should also note that we lost the formation-level number-sync-standbys setting too.

I think in the context of this PR we should move forward with the limitation that we re-join with default replication settings.